openwisp_monitoring/check/classes/iperf.py

```diff
@@ -109,18 +110,63 @@ def validate_params(self, params=None):
             message = '{0}: {1}'.format(message, e.message)
             raise ValidationError({'params': message}) from e
 
-    def check(self, store=True):
-        iperf_config = app_settings.IPERF_CHECK_CONFIG
-        if iperf_config:
-            org_id = str(self.related_object.organization.id)
-            self.validate_params(params=iperf_config[org_id])
+    def _validate_iperf_config(self, org):
+        # if iperf config is present and validate it's params
+        if app_settings.IPERF_CHECK_CONFIG:
+            self.validate_params(
+                params=app_settings.IPERF_CHECK_CONFIG.get(str(org.id))
+            )
 
-        port = self._get_param(
-            'client_options.port', 'client_options.properties.port.default'
-        )
+    def check(self, store=True):
+        lock_acquired = False
+        org = self.related_object.organization
+        self._validate_iperf_config(org)
+        available_iperf_servers = self._get_param('host', 'host.default')
+        if not available_iperf_servers:
+            logger.warning(
+                (
+                    f'Iperf servers for organization "{org}" '
+                    f'is not configured properly, iperf check skipped!'
+                )
+            )
+            return
         time = self._get_param(
             'client_options.time', 'client_options.properties.time.default'
         )
+        # Try to acquire a lock, or put task back on queue
+        for server in available_iperf_servers:
+            server_lock_key = f'ow_monitoring_{org}_iperf_check_{server}'
+            # Set available_iperf_server to the org device
+            lock_acquired = cache.add(
+                server_lock_key,
+                str(self.related_object),
+                timeout=app_settings.IPERF_CHECK_LOCK_EXPIRE,
+            )
+            if lock_acquired:
+                break
+        else:
+            logger.warning(
```

Comment:
let's use `info` instead of warning, since this may happen often on busy systems and I do not see a reason to treat this log line differently than info.

Reply:
I'll update that :+1: